### ISSUE TITLE
PartitionBodyLiteralsTransformer not aggressive enough

### ISSUE THREAD

**Issue by b-scholz** [souffle-lang/souffle#1173]
The following code:
```
.decl B(x:number, y:number)
B(1,1).
.decl A(x:number)
A(1).
A(x+1) :- A(x), x < 100, B(_,1).
.printsize A
```
does not create a nullify for the term ```B(_,1)```.  Assuming that B is a very large relation, this can cost dearly in a recursion like A. 

A better code transformation would be:
```
.decl B(x:number, y:number)
B(1,1).
.decl disconnected1() 
disconnected1() :- B(_,1). 
.decl A(x:number)
A(1).
A(x+1) :- A(x), x < 100, disconnected1.
.printsize A
```

**Comment by azreika** [comment]
This is actually an issue with unnamed variables not being considered in the partitioning process. I'll pull up an extension soon.

**Comment by azreika** [comment]
Solved by #1178.
